### PR TITLE
Success screen displays benefits for creating a user account

### PR DESCRIPTION
**Pull request by Daniela510**

### CODE REVIEW

**Review by mikepattyn — APPROVED** [review]

File: lib/core/constants/benefits.dart [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Map<int, String> benefits = {
```
Comment (phalox) [inline lib/core/constants/benefits.dart:1]: you can change this to a List. Makes more sense than a map I think?

File: lib/core/widgets/buttons/button_square_updt.dart [hunk 2]
```diff
@@ -25,9 +27,7 @@ class SquareButtonE extends StatelessWidget {
           //this doesnt do shit haha. gotta clip out the inkwell somehow else
           borderRadius: BorderRadius.circular(40.0),
           child: InkWell(
-            onTap: () {
-              //To be done
-            },
+            onTap: () => {onClick()},
```
Comment (phalox) [inline lib/core/widgets/buttons/button_square_updt.dart:30]: Are we supposed to have an empty click handler? or is this just turning a tap into a click event? (sorry for stupid question)

File: lib/features/basic_giving_flow/pages/donation_success.dart [hunk 3]
```diff
@@ -8,15 +10,26 @@ import '../../../models/organisation.dart';
...
+class _SuccessDonationPageState extends State<SuccessDonationPage> {
+  int benefitNr = 0;
 
   final NavigationService _navigationService = locator<NavigationService>();
   late final LocalStorageProxy storageProxy = locator<LocalStorageProxy>();
   @override
   void initState() {
     storageProxy.setCompletedOneDonationFlag(true);
+    Random random = Random();
+    benefitNr = random.nextInt(2);
```
Comment (phalox) [inline lib/features/basic_giving_flow/pages/donation_success.dart:31]: I would replace this with [list.shuffle](https://www.fluttercampus.com/guide/215/shuffle-list-dart-flutter/)

Comment (phalox) [reply lib/features/basic_giving_flow/pages/donation_success.dart:31]: And then select the first item

File: lib/features/home_screen/pages/home_screen.dart [hunk 4]
```diff
@@ -66,14 +70,19 @@ class _HomeScreenPageState extends State<HomeScreenPage> {
...
                     children: [
                       SquareButtonE(
+                          onClick: () {
+                            _navigationService.navigateTo(routes.QRScannerPage);
+                          },
                           title: "Scan Qr Code",
                           iconName: "qr_code",
                           background: Theme.of(context).colorScheme.surface),
                       SquareButtonE(
+                          onClick: () => print('non-functional'),
```
Comment (phalox) [inline lib/features/home_screen/pages/home_screen.dart:80]: These are the buttons at the bottom right? would a little toast be better for people testing the app right now? - Not really an action to take, just a thought

**Review by phalox — COMMENTED** [review]
Just one comment, it cleans the code a littlebit, but isn't a blocker if it will take too much time.

File: lib/core/constants/benefits.dart (hunk 1 above)
Comment (pavelpavelko) [reply lib/core/constants/benefits.dart:1]: what do you think about enum in this case?

File: lib/features/home_screen/pages/home_screen.dart (hunk 4 above)
Comment (Daniela510) [reply lib/features/home_screen/pages/home_screen.dart:80]: nope these are buttons on homescreen, the 3 square ones, but there are actually a considerable amount of buttons that are non functional.
Home

**Review by phalox — COMMENTED** [review]
Looks good!

**Pull request merged**